android/src/main/java/com/rivereactnative/RiveReactNativeModule.kt

```diff
@@ -0,0 +1,40 @@
+package com.rivereactnative
+
+import com.facebook.react.bridge.*
+import com.facebook.react.uimanager.UIManagerHelper
+import com.facebook.react.uimanager.UIManagerModule
+
+class RiveReactNativeModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
+  override fun getName() = "RiveReactNativeModule"
+
+  private fun <T> handleState(node: Int, promise: Promise, stateGetter: (RiveReactNativeView) -> T) {
```

Comment:
I created a nice reusable abstraction for Kotlin, which we can definitely do for Swift and JavaScript, as there are places where it's a lot of the same code.

Contributions are welcome if you'd like to clean things up